Modules/IO/TIFF/src/itkTIFFReaderInternal.cxx

```diff
@@ -41,8 +97,34 @@ TIFFReaderInternal::Open(const char * filename)
     return 0;
 #endif
   }
+// Macro added in libtiff 4.5.0
+#if defined(TIFFLIB_AT_LEAST)
 
-  this->m_Image = TIFFOpen(filename, "r");
+  std::unique_ptr<TIFFOpenOptions, tiff_open_options_free> options(TIFFOpenOptionsAlloc());
+  TIFFOpenOptionsSetErrorHandlerExtR(options.get(), itkTIFFErrorHandlerExtR, this);
+  TIFFOpenOptionsSetWarningHandlerExtR(options.get(), itkTIFFWarningHandlerExtR, this);
+  if (silent)
+  {
+    this->m_ErrorSilence = true;
+  }
+  this->m_Image = TIFFOpenExt(filename, "r", options.get());
+
+#else
+  if (silent)
+  {
+
+    // Now check if this is a valid TIFF image
+    TIFFErrorHandler error_save = TIFFSetErrorHandler(nullptr);
```

Comment:
Why do we only override the error handler and not the warning handler (TIFFSetWarningHandler)?

Reply:
This may have been done to support the "CanOpen" function to prevent error messages from non-tiff tiles.